docs/source/how-to/migrations/upgrade-to-16.md

```diff
@@ -0,0 +1,70 @@
+# Upgrading your OAuthenticator to version 16.0
+
+The following section describes what to keep in mind when upgrading to OAuthenticator 16.0.
+
+````{important}
+(token_request)=
+
+```{rubric} The token request
+```
+Whenever _token request_ is used, it refers to the HTTP request exchanging the [oauth code](https://www.rfc-editor.org/rfc/rfc6749#section-1.3.1) for the [access token](https://www.rfc-editor.org/rfc/rfc6749#section-1.4).
+
+This request is sent to the [`token_url`](https://oauthenticator.readthedocs.io/en/latest/reference/api/gen/oauthenticator.oauth2.html#oauthenticator.oauth2.OAuthenticator.token_url) in [`get_token_info()`](https://oauthenticator.readthedocs.io/en/latest/reference/api/gen/oauthenticator.oauth2.html#oauthenticator.oauth2.OAuthenticator.get_token_info) method.
+
+(userdata_request)=
+
+```{rubric} The userdata request
+```
+Whenever _userdata request_ is used, it refers to the HTTP request that's exchanging the the `access token` to get the `userdata`.
+
+This request is sent to [`userdata_url`](https://oauthenticator.readthedocs.io/en/latest/reference/api/gen/oauthenticator.oauth2.html#oauthenticator.oauth2.OAuthenticator.userdata_url) in [`token_to_user`](https://oauthenticator.readthedocs.io/en/latest/reference/api/gen/oauthenticator.oauth2.html#oauthenticator.oauth2.OAuthenticator.get_token_info).
```

Comment:
It seems like there are triple-backticks that should encapsulate this section, but are instead just start-end on line 16-17

Reply:
It is my understanding that the syntax of a [`rubric` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-rubric) is to only encapsulate a title. 

I believe other content is not allowed inside the directive. And this is the error got when trying to do it.

```
 ERROR: Directive 'rubric': No content permitted
 ```

### Some context about the `rubric` directive and how I found out about it

The reason why I used this directive was because I wanted to have a heading and to be able to cross-reference it from different places in the documentation.

But a heading inside an ```{admonition} is not permitted and [they get turned into rubrics](https://myst-parser.readthedocs.io/en/latest/develop/_changelog.html#internal-improvements).

So, I thought it's best to be explicit instead of relying on the default behavior.

Do you agree with this reasoning @consideRatio ?

Reply:
![image](https://user-images.githubusercontent.com/3837114/213675001-eabd0846-8eb8-4750-be70-e29fe13770c7.png)

Ah, I had never used this and benefited from your explanations and links. This looks smart to me!!